src/coord/src/catalog/storage.rs

```diff
@@ -481,6 +517,36 @@ impl Transaction<'_> {
         }
     }
 
+    pub fn insert_timestamp(
+        &self,
+        source_id: &GlobalId,
+        partition_id: &str,
+        timestamp: Timestamp,
+        offset: i64,
+    ) -> Result<(), Error> {
+        match self
+            .inner
+            .prepare_cached(
+                "INSERT OR IGNORE INTO timestamps (sid, pid, timestamp, offset) VALUES (?, ?, ?, ?)",
```

Comment:
document why its insert or ignore instead of just inserts -- spoiler - byo ts bindings are multicast to all workers and all of them want to tell the coord about it.